feat(ast, parser, linter, codegen, formatter)!: rename `CommentKind::Block` to `CommentKind::SinglelineBlock`

Dunqing opened the pull request:
https://github.com/oxc-project/oxc/pull/16479 introduced a new kind called `CommentKind::MultilineBlock`, which means a block comment that has at least one line break. Therefore, `CommentKind::Block` means it is a single-line block.

In this PR, rename `CommentKind::Block` to `CommentKind::SinglelineBlock` to make the kind name more descriptive.

overlookmotel commented:
Also I've marked this as a breaking change, as it is a change which may affect downstream crate consumers.

Dunqing committed:
feat(ast, parser, linter, codegen, formatter)!: rename `CommentKind::Block` to `CommentKind::SinglelineBlock` (#16501)

https://github.com/oxc-project/oxc/pull/16479 introduced a new kind called `CommentKind::MultilineBlock`, which means a block comment that has at least one line break. Therefore, `CommentKind::Block` means it is a single-line block.

In this PR, rename `CommentKind::Block` to `CommentKind::SinglelineBlock` to make the kind name more descriptive.
